crates/viewer/re_view_spatial/tests/transform_hierarchy.rs

```diff
@@ -0,0 +1,271 @@
+                RowId::new(),
+                [(timeline_step, 5)],
+                &re_types::archetypes::Transform3D::from_rotation(
+                    re_types::components::RotationAxisAngle::new(
+                        (0.0, 1.0, 0.0),
+                        re_types::datatypes::Angle::from_degrees(90.0),
+                    ),
+                ),
+            )
+        });
+
+        path = path.join(&"rotate_quat".into());
+        test_context.log_entity(path.clone(), |builder| {
+            builder.with_archetype(
+                RowId::new(),
+                [(timeline_step, 6)],
+                &re_types::archetypes::Transform3D::from_rotation(
+                    // -45 degrees around the y axis.
+                    // Via https://www.andre-gaschler.com/rotationconverter/
+                    re_types::components::RotationQuat(re_types::datatypes::Quaternion::from_xyzw(
+                        [0.0, -0.3826834, 0.0, 0.9238796],
+                    )),
+                ),
+            )
+        });
+
+        path = path.join(&"rotate_mat3x3".into());
+        test_context.log_entity(path.clone(), |builder| {
+            builder.with_archetype(
+                RowId::new(),
+                [(timeline_step, 7)],
+                // -45 degrees around the y axis.
+                // Via https://www.andre-gaschler.com/rotationconverter/
+                &re_types::archetypes::Transform3D::from_mat3x3([
+                    [0.7071069, 0.0000000, -0.7071066],
+                    [0.0000000, 1.0000000, 0.0000000],
+                    [0.7071066, 0.0000000, 0.7071069],
+                ]),
+            )
+        });
+
+        // Add the Rerun asset at the end of the hierarchy.
+        // (We're using a 3D model because it's easier to see the effect of arbitrary transforms here!)
+        {
+            let workspace_dir = std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+            let workspace_dir = workspace_dir
+                .parent()
+                .and_then(|p| p.parent())
+                .and_then(|p| p.parent())
+                .unwrap();
+            let obj_path = workspace_dir.join("tests/assets/rerun.obj");
+
+            path = path.join(&"asset".into());
+            test_context.log_entity(path.clone(), |builder| {
+                builder.with_archetype(
+                    RowId::new(),
+                    [(timeline_step, 0)],
+                    &re_types::archetypes::Asset3D::from_file(&obj_path).unwrap(),
+                )
+            });
+        }
+    }
+
+    let view_id = setup_blueprint(&mut test_context);
+
+    run_view_ui_and_save_snapshot(
+        &mut test_context,
+        timeline_step,
+        view_id,
+        "transform_hierarchy",
+        // TODO(#8924): A lot of pixels won't pass the diff because of anti-aliasing.
+        // By bumping the resolution, we lessen the weight of edges.
+        egui::vec2(300.0, 150.0) * 3.0,
```

Comment:
intuitively I'd say we can't make the picture all that much smaller without risking the test mesh not to be significant, but zooming in a bit would help a lot?